Build failing on NetBSD

Probably a non-goal to make it work on NetBSD, but for the record:

```
#=== ERROR while compiling async_ssl.v0.13.0 ==================================#
# context     2.0.6 | netbsd/evbarm | ocaml-system.4.09.0 | https://opam.ocaml.org#0c362d72
# path        ~/.opam/ocaml-system/.opam-switch/build/async_ssl.v0.13.0
# command     ~/.opam/ocaml-system/bin/dune build -p async_ssl -j 6
# exit-code   1
# env-file    ~/.opam/log/async_ssl-12573-20b9b8.env
# output-file ~/.opam/log/async_ssl-12573-20b9b8.out
### output ###
#         bash bindings/discover.exe (exit 1)
# (cd _build/default/bindings && /usr/pkg/bin/bash -e -u -o pipefail -c 'gcc -O2 -fno-strict-aliasing -fwrapv -fPIC $(< openssl-ccopt) discover.c $(< openssl-cclib) -ldl -o discover.exe')
# ld: cannot find -ldl
```

NetBSD does not do have configurable dynamic linking like Linux…

--- a/bindings/dune
+++ b/bindings/dune
@@ -10,7 +10,7 @@
  (deps discover.c openssl-ccopt openssl-cclib)
  (action
   (bash
-    "%{cc} $(< openssl-ccopt) discover.c $(< openssl-cclib) -ldl -o discover.exe")))
+    "%{cc} $(< openssl-ccopt) discover.c $(< openssl-cclib) $([[ $(uname) == *BSD ]] || echo -n -ldl) -o discover.exe")))
 
 (rule
  (targets config.h)

